### PR TITLE
Mark some enums that end up in the public Rust API as non_exhaustive

### PR DESCRIPTION
**Pull request by tronical**
This allows adding new variants in minor releases without breaking
source compatibility.

cc #431

### CODE REVIEW

**Comment by tronical** [comment]
[`DiagnosticsLevel`](https://sixtyfps.io/snapshots/master/docs/rust/sixtyfps_interpreter/enum.diagnosticlevel) is not marked as non_exhaustive. We match this ourselves in the lsp, for example - making it non_exhaustive would mean we don't get an error ourselves anymore in our code base.

Three options:

    (1) Accept that (and use that warning once it stabilises)
    (2) Separate public and private API (a bit of work, but ok)
    (3) Keep it non_exhaustive


I'm leaning towards (1). What do you think?

File: sixtyfps_runtime/interpreter/api.rs [hunk 1]
```diff
@@ -21,6 +21,7 @@ pub use sixtyfps_corelib::window::api::Window;
 /// the contained values.
 #[derive(Debug, Copy, Clone, PartialEq)]
 #[repr(i8)]
+#[non_exhaustive]
```
Comment (ogoffart) [inline sixtyfps_runtime/interpreter/api.rs:24]: maybe the `Other` should be `#[doc(hidden)]` and handle with the `_` pattern anyway?

File: sixtyfps_runtime/interpreter/api.rs [hunk 2]
```diff
@@ -1059,6 +1063,7 @@ pub enum SetCallbackError {
 
 /// Error returned by [`ComponentInstance::invoke_callback`]
 #[derive(Debug, Clone, Copy, PartialEq, Eq, thiserror::Error)]
+#[non_exhaustive]
```
Comment (ogoffart) [inline sixtyfps_runtime/interpreter/api.rs:1066]: In a folloup patch, we can add the error if arguments type are incorrect

**Review by ogoffart — APPROVED** [review]

**Comment by ogoffart** [comment]
> DiagnosticsLevel is not marked as non_exhaustive.

Option 1.  accept that.

File: sixtyfps_runtime/interpreter/api.rs (hunk 1 above)
Comment (tronical) [reply sixtyfps_runtime/interpreter/api.rs:24]: 👍

File: sixtyfps_runtime/interpreter/api.rs (hunk 2 above)
Comment (tronical) [reply sixtyfps_runtime/interpreter/api.rs:1067]: 👍 Please do

**Pull request merged**